src/utility_components/NumericField.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { forceNumber } from '../utils';
+
+export class NumericField extends Component {
+    constructor(props) {
+        super(props);
+
+        this.onFocus = this.onFocus.bind(this);
+        this.onBlur = this.onBlur.bind(this);
+        this.onChange = this.onChange.bind(this);
+
+        this.state = {
+            value: this.props.value,
+            fieldValue: this.props.value,
+        };
+    }
+    static getDerivedStateFromProps(props, state) {
```

Comment:
The API docs give you fair warning re this lifecycle method: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops
I used it here because this component has only two objects in its state, it keeps the interface as much like an input as possible, and the state objects are very unlikely to grow beyond these two.  If I didn't use this method, I could trigger a reinstantiation of the component by passing in a `key` prop: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html?no-cache=1#recommendation-fully-uncontrolled-component-with-a-key

Reply:
I don't see why this is necessary  - I'm seeing the same behavior with or without this method.

Reply:
On `/ols-regression` try moving the slider after entering a value. W/o this method the numeric field should fail to update.